src/ophys_etl/modules/segmentation/detect/feature_vector_rois.py

```diff
@@ -552,9 +551,14 @@ def select_pixels(self) -> bool:
         # take the median of the n_roi nearest background distances;
         # hopefully this will limit the effect of outliers
         d_bckgd = np.sort(self.feature_distances[:, background_mask], axis=1)
-        d_bckgd = np.median(d_bckgd[:, :n_roi], axis=1)
+        if n_roi > 20:
+            d_bckgd = d_bckgd[:, :n_roi]
+
+        mu_d_bckgd = np.mean(d_bckgd, axis=1)
+        std_d_bckgd = np.std(d_bckgd, axis=1, ddof=1)
+        z_score = (d_roi-mu_d_bckgd)/std_d_bckgd
 
-        valid = d_bckgd > 2*d_roi
+        valid = z_score <= -2.0
```

Comment:
wouldn't just:
```
valid = mu_d_background >= 2.0 * d_roi
```
give the same thing, in which case the substantive change here is changing from median to mean, right?

Reply:
In order to be valid, a pixel has to satisfy
```
(d_roi-mu_d_bckgd)/std_d_bckgd <= -2.0
```
which can be algebraically manipulated to
```
d_roi - mu_d_bckgd <= -2*std_d_bckgd
d_roi <= mu_d_bckgd - 2*std_d_bckgd
```
or
```
mu_d_bckgd >= d_roi + 2*std_d_bckgd
```
or am I missing something?

Reply:
ah, you're right...
can you update part (4) of the docstring?